wrangler-service/src/main/java/io/cdap/wrangler/service/schema/SchemaViewerHandler.java

```diff
@@ -0,0 +1,219 @@
+  /**
+   * Returns a listing of the meta data for all available standards that have schema files.
+   * By default excludes the JsonSchema definitions. If the JsonSchema is required, the standard
+   * specific endpoint (../schemas/{id}) should be used.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   * @param namespace Namespace of request.
+   */
+  @GET
+  @Path("contexts/{context}/schemaviewer/schemas")
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  public void getSchemas(HttpServiceRequest request, HttpServiceResponder responder,
+      @PathParam("context") String namespace) {
+    respond(request, responder, namespace, ns -> TransactionRunners.run(getContext(), context -> {
+      SchemaFileStore store = SchemaFileStore.get(context);
+      List<SchemaListingResponse> listing = new ArrayList<>();
+      for (Map.Entry<String, SchemaFile.MetaData> entry :
+          store.listSchemaFileMetadata().entrySet()) {
+        SchemaFile.MetaData metaData = entry.getValue();
+        listing.add(new SchemaListingResponse(entry.getKey(),
+            metaData.getStandard(),
+            metaData.getVersion()));
+      }
+      return new ServiceResponse<>(listing);
+    }));
+  }
+
+  /**
+   * Returns the JsonSchema for the standard.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   * @param namespace Namespace of request.
+   * @param id The id of the standard returned by the .../schemas listing.
+   */
+  @GET
+  @Path("contexts/{context}/schemaviewer/schemas/{id}")
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  public void getSchema(HttpServiceRequest request, HttpServiceResponder responder,
+      @PathParam("context") String namespace, @PathParam("id") String id) {
+    respond(request, responder, namespace, ns -> TransactionRunners.run(getContext(), context -> {
+      SchemaFileStore store = getStore(context, id);
+      return new ServiceResponse<>(store.readSchemaFileDescriptor(id));
+    }));
+  }
+
+  /**
+   * Returns all of the top level entities associated with a specific standard.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   * @param namespace Namespace of request.
+   * @param id The id of the standard returned by the .../schemas listing.
+   */
+  @GET
+  @Path("contexts/{context}/schemaviewer/schemas/{id}/entities")
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  public void getSchemaEntities(HttpServiceRequest request, HttpServiceResponder responder,
+      @PathParam("context") String namespace, @PathParam("id") String id) {
+    respond(request, responder, namespace, ns -> TransactionRunners.run(getContext(), context -> {
+      SchemaFileStore store = getStore(context, id);
+      List<String> entities = new ArrayList<>();
+      Discriminator discriminator = store.readDiscriminator(id);
```

Comment:
handlers should have as little business logic as possible and be primarily concerned with handling errors, deserializing request bodies, etc.

The SchemaDescriptor class should have something like a `getTopLevelFields()` method that this handler just calls. A Json schema would implement this differently than an Avro schema.